### PR TITLE
 front: refactorization and implementation of types for trainschedule import from json files

### LINKED ISSUE #3547: Display PTO imported data on OSRD
### Description and goal

Being able to generate a time table from PTO data.

### Acceptance criteria

- [ ] Read PTO data from a locally stored Json.
- [ ] Display the train list 
- [ ] Display the timetable

### PR DESCRIPTION
**Pull request by Math-R**
close #3547

### CODE REVIEW

File: front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleConfig.tsx [hunk 1]
```diff
@@ -8,9 +8,18 @@ import { setFailure } from 'reducers/main';
...
 import { formatIsoDate } from 'utils/date';
+import UploadFileModal from 'applications/customget/components/uploadFileModal';
+import { ModalContext } from 'common/BootstrapSNCF/ModalSNCF/ModalProvider';
+import { TrainSchedule } from 'applications/operationalStudies/views/types';
 
-export default function ImportTrainScheduleConfig(props) {
-  const { setConfig } = props;
+interface ImportTrainScheduleConfigProps {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  setConfig: (config: any) => void;
```
Comment (nicolaswurtz) [inline front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleConfig.tsx]: Use a specific type

File: front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleTrainDetail.tsx [hunk 2]
```diff
@@ -29,19 +32,21 @@ export default function ImportTrainScheduleTrainDetail({ trainData, idx, rolling
...
-          {trainData.num}
-          <span className="import-train-schedule-traindetail-activity">{trainData.origine}</span>
+          {trainData.trainNumber}
+          <span className="import-train-schedule-traindetail-activity">
+            {trainData.departure.name}
+          </span>
         </span>
         <span className="import-train-schedule-traindetail-startend">
-          {trainData.debut} - {trainData.fin}
+          {trainData.departureTime.slice(-8)} - {trainData.arrivalTime.slice(-8)}
```
Comment (nicolaswurtz) [inline front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleTrainDetail.tsx:47]: Great !

File: front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleTrainDetail.tsx [hunk 3]
```diff
@@ -50,23 +55,28 @@ export default function ImportTrainScheduleTrainDetail({ trainData, idx, rolling
             </span>
           </LazyLoadComponent>
         )}
-        <span className="import-train-schedule-traindetail-transilien">
-          {trainData.num_transilien}
+        <span className="import-train-schedule-traindetail-rollingstock-name">
+          {trainData.rollingStock}
+          {trainData.numTransilien && (
+            <span className="import-train-schedule-traindetail-transilien">
+              {trainData.numTransilien}
```
Comment (nicolaswurtz) [inline front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleTrainDetail.tsx]: Use `transilienName` instead, no ?

File: front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleTrainsList.tsx [hunk 4]
```diff
@@ -2,31 +2,45 @@ import React, { useEffect, useState, useMemo } from 'react';
 import { useTranslation } from 'react-i18next';
 import PropTypes from 'prop-types';
```
Comment (nicolaswurtz) [inline front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleTrainsList.tsx]: Remove the line

File: front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleTrainsList.tsx [hunk 5]
```diff
@@ -2,31 +2,45 @@ import React, { useEffect, useState, useMemo } from 'react';
...
       {isSearching ? <Loader position="center" /> : t('noResults')}
     </h1>
   );
 }
 
-export default function ImportTrainScheduleTrainsList(props) {
+type ImportTrainScheduleTrainsListProps = {
+  config?: Object;
+  trainsList?: TrainSchedule[];
+  rollingStockDB: any[];
```
Comment (nicolaswurtz) [inline front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleTrainsList.tsx]: There is a dedicated type from RTK

File: front/src/applications/operationalStudies/views/types.ts
Comment (nicolaswurtz) [inline front/src/applications/operationalStudies/views/types.ts:1]: Please use ../consts.ts where other types lies (or split files at the top of operationalStudies/)

**Review by nicolaswurtz — CHANGES_REQUESTED** [review]

File: front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleTrainDetail.tsx (hunk 3 above)
Comment (Math-R) [reply front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleTrainDetail.tsx]: Maybe, i keep it like you named it before but i can make the change

File: front/src/applications/operationalStudies/views/types.ts
Comment (Math-R) [reply front/src/applications/operationalStudies/views/types.ts:1]: Done !

File: front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleTrainsList.tsx (hunk 5 above)
Comment (Math-R) [reply front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleTrainsList.tsx]: Done !

File: front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleTrainDetail.tsx (hunk 3 above)
Comment (Math-R) [reply front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleTrainDetail.tsx]: Done !

File: front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleConfig.tsx (hunk 1 above)
Comment (Math-R) [reply front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleConfig.tsx]: Done !

**Review by nicolaswurtz — APPROVED** [review]

**Pull request merged**